CMakeLists.txt

```diff
@@ -46,20 +48,17 @@ if(VULKAN_FOUND)
                     "$ENV{VULKAN_SDK}/bin"
                     )
     endif()
-    if(DEFINED Vulkan_dxc_EXECUTABLE)
-        message("Found DirectX Shader Compiler under ${Vulkan_dxc_EXECUTABLE}")
-    elseif()
-        message("Couldn't find DirectX Shader Compiler executable, make sure it is present in Vulkan SDK or add it manually via Vulkan_dxc_EXECUTABLE env variable")
+    if(Vulkan_dxc_EXECUTABLE)
+        message(STATUS "Found DirectX Shader Compiler under ${Vulkan_dxc_EXECUTABLE}")
+    else()
+        message(STATUS "Couldn't find DirectX Shader Compiler executable, make sure it is present in Vulkan SDK or add it manually via Vulkan_dxc_EXECUTABLE cmake variable. HLSL shaders won't be compiled.")
+        unset(Vulkan_dxc_EXECUTABLE CACHE)
```

Comment:
Why would we unset this if it's not found?

Reply:
I did this for performance reasons.  See the following code snippet from **sample_helper.cmake**:

```
    if(DEFINED Vulkan_dxc_EXECUTABLE AND DEFINED SHADERS_HLSL)
        compile_hlsl_shaders(
            SHADERS_HLSL ${TARGET_SHADERS_HLSL}
            DXC_ADDITIONAL_ARGUMENTS ${TARGET_DXC_ADDITIONAL_ARGUMENTS}
        )
    endif()
```

When dxc is **not** found, `${Vulkan_dxc_EXECUTABLE}` will equal `Vulkan_dxc_EXECUTABLE-NOTFOUND`, which is **DEFINED** in cmake semantics.  If I did not unset the variable, the logic above will run erroneously and slow down/timeout on a missing dxc executable.  A different solution might change the above code to either:

`if(Vulkan_dxc_EXECUTABLE` ... or `if(Vulkan_dxc_exe_FOUND ...`

I did not do either of these since I wanted to minize the amount of code I touched in the PR.  If you don't like the `unset()`, then I could fix the problem within **sample_helper.cmake** as above.  Please advise.

Reply:
It's probably better to go ahead and change it to if(Vulkan_dxc_exe_FOUND to be more idomatic as this is a sample of best practices repo then that path does hold weight with me.

Reply:
OK thanks very much for the feedback.  I will make the change, retest, and submit.

Reply:
Fixed in https://github.com/KhronosGroup/Vulkan-Samples/pull/1084/commits/1e3e85b9ab95a2e04b2d0180452ce9b68f509bff (`Vulkan_dxc_EXECUTABLE` used instead of `DEFINED Vulkan_dxc_EXECUTABLE`).

Cannot use `Vulkan_dxc_exe_FOUND` here due to `find_program()` fallback in previous review point.